examples/webtransport_server.rs

```diff
@@ -0,0 +1,328 @@
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+/// This method will echo all inbound datagrams, unidirectional and bidirectional streams.
+#[tracing::instrument(level = "info", skip(session))]
+async fn handle_session_and_echo_all_inbound_messages<C>(
+    session: WebTransportSession<C>,
+) -> anyhow::Result<()>
+where
+    C: 'static + Send + h3::quic::Connection,
+    <C::SendStream as h3::quic::SendStream>::Error:
+        'static + std::error::Error + Send + Sync + Into<std::io::Error>,
+    C::SendStream: Unpin,
+{
+    let open_bi_timer = sleep(Duration::from_millis(10000));
+    pin!(open_bi_timer);
+    let session_id = session.session_id();
+
+    loop {
+        tokio::select! {
+            datagram = session.accept_datagram() => {
+                let datagram = datagram?;
+                if let Some((_, datagram)) = datagram {
+                    tracing::info!("Responding with {datagram:?}");
+                    // Put something before to make sure encoding and decoding works and don't just
+                    // pass through
+                    let mut resp = BytesMut::from(&b"Response: "[..]);
+                    resp.put(datagram);
+
+                    session.send_datagram(resp).unwrap();
+                    tracing::info!("Finished sending datagram");
+                }
+            }
+            uni_stream = session.accept_uni() => {
+                let (id, mut stream) = uni_stream?.unwrap();
+                tracing::info!("Received uni stream {:?} for {id:?}", stream.recv_id());
+
+                let mut message = BytesMut::new();
+                while let Some(bytes) = poll_fn(|cx| stream.poll_data(cx)).await? {
+                    tracing::info!("Received chunk {:?}", &bytes);
+                    message.put(bytes.clone());
+                }
+
+                tracing::info!("Got message: {message:?}"); 
+                let mut send = session.open_uni(session_id).await?;
+                let message = message.to_vec();
+                let mut message = message.as_slice();
+               
+                // TODO: move this into the library
+                // send stream type
+                {
+                    let mut data = [0u8; 8];
+                    let mut buf = octets::OctetsMut::with_slice(&mut data);
+                    buf.put_varint(WEBTRANSPORT_UNI_STREAM_TYPE).unwrap();
+                    send.write(&mut buf.buf()).await.context("Failed to respond")?;
+                }
+
+                // Send session id
+                {
+                    let mut data = [0u8; 8];
+                    let mut buf = octets::OctetsMut::with_slice(&mut data);
+                    buf.put_varint(session_id.0).unwrap();
+                    send.write(&mut buf.buf()).await.context("Failed to respond")?;
+                }
+                // Send actual data.
+                send.write(&mut message).await.context("Failed to respond")?;
```

Comment:
@ten3roberts can you spot what I am doing wrong here? 

I am trying to send the stream type followed by the session id, then the data per the spec.

This is the quiche equivalent for sending the stream type: https://github.com/cloudflare/quiche/blob/330b24e2f01ac4e16313154f753a9228cd9e1061/quiche/src/h3/mod.rs#L2082-L2120